hack/cluster.sh

```diff
@@ -17,33 +17,41 @@
 # Copyright 2022 The Kepler Contributors
 #
 
-set -e
+set -eu -o pipefail
 
+# config
+declare -r VERSION=${VERSION:-v0.0.1}
+declare -r CLUSTER_PROVIDER=${CLUSTER_PROVIDER:-kind}
+
+# constants
 PROJECT_ROOT="$(git rev-parse --show-toplevel)"
 declare -r PROJECT_ROOT
 declare -r TMP_DIR="$PROJECT_ROOT/tmp"
-declare -r CLUSTER_PROVIDER=${CLUSTER_PROVIDER:-kind}
-declare -r VERSION=${VERSION:-v0.0.1}
 
-function main() {
+git_checkout() {
+	local version="$1"
+
+	[[ -d local-dev-cluster ]] || {
+		echo "INFO: downloading local-dev-cluster - version $VERSION"
+		git clone -b "$version" https://github.com/sustainable-computing-io/local-dev-cluster.git --depth=1
+	}
 
-    if [ ! -d "${TMP_DIR}" ]
-    then
-        mkdir "${TMP_DIR}"
-    fi
+	cd local-dev-cluster
+	git reset --hard "$version"
```

Comment:
If the folder is already exists (i.e. already cloned, git reset to the VERSION instead. 

> Also, this blocks anyone who wants to run a customized version of local-dev-cluster

can you please expand further how this block?

Reply:
Let's say that folder already exists and I made some modifications to any of the provider's scripts. In that case when I do `make cluster-up` the line `git reset --hard "$version"` will remove my changes.

Reply:
For local modifications, I would recommend that you clone / copy the local-dev-cluster and use that to setup. We can check is the git status is dirty and bail out if it isn't clean. I have made that change now so that local modifications aren't lost by mistake.